### DISCUSSION TITLE
Can server middleware be scoped to specific  `~/server/api/` routes?

### DISCUSSION THREAD

**Discussion by john-prutton** [nuxt/nuxt#16916] [Questions] [answered]
Hi. I want to create `~/server/middleware/auth.ts` to check the authentication of requests to certain routes in my api. For instance, a request to `~/server/api/user/login.post.ts` would not need to be authenticated, while a request to `~/server/api/workouts/create.post.ts` would need to be authenticated.

It would be great if it worked similar to client-side route middleware, that can be enabled with `definePageMeta({middleware: "auth"})`.

I've had a look at the [Nuxt server middleware](https://v3.nuxtjs.org/guide/features/server-routes#server-middleware) and  [Nitro](https://nitro.unjs.io/guide/introduction/routing) docs, as well as the respective Github discussions, but didn't manage to find anything.

**Should I be using middleware for this, or is there another way to share code across api routes?**

Thanks for any help. :]

**Comment by wobsoriano** [comment]
Yep, server middleware is what you need.

**Reply by john-prutton** [reply]
I am looking for a way to scope server middleware to specific routes. I don't want a specific middleware to be run on all api routes.

**Comment by mihail727** [comment, marked as answer]
you can manually specify which requests to protect
```ts
export default defineEventHandler(async (event) => {
	const targetUrl = event.req.url;
	const securedUrls = ['/api/test'];
	const bypass = securedUrls.some((url) => targetUrl.includes(url));
	if (bypass) {
		// some code if it target route
	}
});

```

**Reply by john-prutton** [reply]
Thanks for the suggestion. I can see something like this working to whitelist a few api routes while protecting the rest. I'll mark this as the answer, although I am considering using a composable at the top of (about half) my api routes, `useApiAuthorization(even)`, which would grab the auth token and return a user from the database - throwing a `401 UNATHORIZED` HTTP response with `sendError` and `createError` if there was no valid auth token found.
